test/dynamo/test_structured_trace.py

```diff
@@ -142,7 +142,7 @@ def test_schedule(self):
 {"inductor_output_code": {"filename": "FILENAME"}, "frame_id": 0, "frame_compile_id": 0, "attempt": 0, "has_payload": "HASH"}
 {"dynamo_guards": {}, "frame_id": 0, "frame_compile_id": 0, "attempt": 0, "has_payload": "HASH"}
 {"dynamo_cpp_guards_str": {}, "frame_id": 0, "frame_compile_id": 0, "attempt": 0, "has_payload": "HASH"}
-{"compilation_metrics": "METRICS", "frame_id": 0, "frame_compile_id": 0, "attempt": 0}
+{"fwd_compilation_metrics": "METRICS", "frame_id": 0, "frame_compile_id": 0, "attempt": 0}
```

Comment:
You're still egregiously breaking BC for no good reason

Reply:
Because we collect these metrics only for fwd graphs, ```BwdCompilationMetrics``` is used to store compilation metrics for bwd graphs. I'm ok to keep ```CompilationMetrics``` and add ```BwdCompilationMetrics```, but not sure if this would confuse others who look at these metrics.

Reply:
You're welcome to rename the struct, but don't rename the JSON key name, that's BC surface

Reply:
I'll keep the original ```CompilationMetrics``` for fwd and add ```BwdCompilationMetrics``` for bwd.